lib/perl/Genome/Model/Tools/Picard/Base.pm

```diff
@@ -128,18 +128,26 @@ sub _redirects {
     return sprintf(">> %s", $self->log_file) if $self->log_file;
 }
 
-sub build_cmdline_string {
-    my ($self, $cmd) = shift;
+sub build_cmdline_list {
+    my $self = shift;
     my $jvm_options = $self->additional_jvm_options || '';
 
-    my $java_vm_cmd = sprintf '%s -Xmx%dm -XX:MaxPermSize=%dm %s -cp /usr/share/java/ant.jar:%s %s %s',
+    return (
         $self->java_interpreter,
-        int(1024 * $self->maximum_memory),
-        $self->maximum_permgen_memory,
-        $jvm_options,
-        $self->_jar_path,
+        sprintf('-Xmx%dm', int(1024 * $self->maximum_memory)),
+        sprintf('-XX:MaxPermSize=%dm', $self->maximum_permgen_memory),
+        split(' ', $jvm_options),
```

Comment:
Actually, `shellwords` might be more harmful due to lack of symmetry with the `join`.